src/linker/Linker/DirectoryAssemblyResolver.cs

```diff
@@ -40,7 +45,34 @@ protected AssemblyDefinition GetAssembly (string file, ReaderParameters paramete
 			if (parameters.AssemblyResolver == null)
 				parameters.AssemblyResolver = this;
 
-			return ModuleDefinition.ReadModule (file, parameters).Assembly;
+			FileStream fileStream = null;
+			MemoryMappedFile mappedFile = null;
+			MemoryMappedViewStream viewStream = null;
+			try {
+				// Create stream because CreateFromFile(string, ...) uses FileShare.None which is too strict
+				fileStream = new FileStream (file, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, false);
+				mappedFile = MemoryMappedFile.CreateFromFile (
+					fileStream, null, fileStream.Length, MemoryMappedFileAccess.Read, HandleInheritability.None, true);
+				viewStream = mappedFile.CreateViewStream (0, 0, MemoryMappedFileAccess.Read);
+
+				AssemblyDefinition result = ModuleDefinition.ReadModule (viewStream, parameters).Assembly;
+
+				AssemblyToPath.Add (result, file);
+
+				viewStreams.Add (viewStream);
+
+				// We transferred the ownership of the viewStream to the collection.
+				viewStream = null;
```

Comment:
why is this needed?

Reply:
It's needed because we want to `Dispose` the `viewStream` if `ModuleDefinition.ReadModule` above failed for any reason - such assembly is unusable and we might as well throw away the mapping now. If accessing the assembly from the module succeeded, we do not want to dispose it because it's used by Cecil and the `viewStreams` collection now manages the lifetime.

It's also a reason why I'm not using a `using` - the `viewStream` handling is odd. If you see a better way to do this, I can change it.

Reply:
You might consider using `using` for `fileStream` and `mappedFile` and keep the only `viewStream` in custom finally to make the logic/flow more obvious